app/src/main/java/com/mymeetings/android/view/viewModels/CalendarEventsViewModel.kt

```diff
@@ -0,0 +1,69 @@
+package com.mymeetings.android.view.viewModels
+
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModel
+import com.mymeetings.android.model.CalendarEvent
+import com.mymeetings.android.model.managers.CalendarEventAlertManager
+import com.mymeetings.android.model.managers.CalendarEventsSyncManager
+import com.mymeetings.android.utils.ClockUtils
+import com.mymeetings.android.view.uiModels.CalendarEventAlertUIModel
+import kotlinx.coroutines.*
+
+class CalendarEventsViewModel(
+    private val calendarEventsSyncManager: CalendarEventsSyncManager,
+    private val calendarEventAlertManager: CalendarEventAlertManager,
+    private val clockUtils: ClockUtils
+) : ViewModel() {
+
+    private val backgroundScope = CoroutineScope(Dispatchers.IO)
+
+    val calendarEventAlertsLiveData = MutableLiveData<List<CalendarEventAlertUIModel>>()
+
+    private val observer = Observer<List<CalendarEvent>> { calendarEvents ->
+        calendarEventAlertsLiveData.postValue(calendarEvents.map { calendarEvent ->
+            computeCalendarEventUIModel(calendarEvent)
+        })
+    }
+
+    init {
+        calendarEventsSyncManager.calendarEventsLiveData.observeForever(observer)
+    }
+
+    private fun computeCalendarEventUIModel(calendarEvent: CalendarEvent): CalendarEventAlertUIModel {
```

Comment:
This should be moved to the `MapperExtensions`, otherwise the ViewModel will become bloated and the code has potential of being duplicated in the future.